barnold/addon_preferences.py

```diff
@@ -19,36 +22,98 @@ class ArnoldAddonPreferences(AddonPreferences):
 
     def draw(self, context):
         layout = self.layout
-        layout.label(text="IMPORTANT NOTICE:")
-        layout.label(text="if you have an ARNOLD_HOME environment set,it will \
-        override whatever setting you input here.")
+        column = layout.column(align=True)
+        column.label(text="IMPORTANT NOTICE:")
+        column.label(text="if you have an ARNOLD_HOME environment set,")
+        column.label(text="it will be the fallback to whatever you input here.")
         layout.prop(self, "arnold_path")
 
-# Registration
-
 
-def register():
-    bpy.utils.register_class(ArnoldAddonPreferences)
+# Registration
+@persistent
+def load_handler(dummy):
+    p = bpy.context.preferences.addons[__package__].preferences
+    if add_arnold_path(p):
+        try:
+            register_arnold()
+        except Exception as e:
+            print("LOAD HANDLER: Cannot register arnold")
+            print(e)
+
+def add_arnold_path(prefs):
     try:
-        pth = os.environ["ARNOLD_HOME"]
-        print("ARNOLD_HOME env found")
-
+        import arnold
+        return True
     except:
-        print("ARNOLD_HOME env not found, using the preferences.")
-        prefs = bpy.context.user_preferences.addons[__package__].preferences
-        pth = prefs.arnold_path
-
-    print("Setting Arnold path to: {}".format(pth))
-
-    pth = os.path.join(pth, "python")
-
-    if pth not in sys.path:
-        sys.path.append(pth)
-
+        if prefs:
+            pth = prefs.arnold_path
+            pth = os.path.join(pth,'python')
+            if pth:
+                if os.path.isdir(pth):
+                    if pth not in sys.path:
+                        print("Appending {} to sys.path".format(pth))
+                        sys.path.append(pth)
+                    return True
+                else:
+                    print("Path set is not a directory, aborting.")
+                    return False
+        
+        print("Using Environment Fallback")
+        pth = os.environ["ARNOLD_HOME"]
```

Comment:
Instead of using keyed lookup of the `os.environ` dict, you should use: 
`pth = os.environ.get("ARNOLD_HOME", None)`. This will set the key to the default value `None` if not found, instead of throwing a `KeyError`. This way we can remove the try/except block all together.

Reply:
of course !! I'll work on it now. it was a draft to test the mechanism. It seams to be pretty steady. I'll try a few test cases and update the PR later today.